### PR TITLE
Collect LSTM quant model scales and save LSTM int8

### PR DESCRIPTION
**Pull request by lidanqing-vv**
### PR types
Function optimization

### PR changes
Others

### Describe
Add fc lstm fuse pass, collect quant2 lstm scales and save lstm int8 model test

### CODE REVIEW

**Review by wozna — DISMISSED** [review]
Well done. LGTM

File: paddle/fluid/framework/ir/graph_pattern_detector.cc [hunk 1]
```diff
@@ -2253,11 +2253,13 @@ PDNode *patterns::QuantizePlacement::operator()(
...
-  }
-  auto *op = pattern->NewNode(op_repr())->assert_is_ops(supported_op_types);
-  return op;
+  "multi_gru", "fusion_lstm"
+});
+if (!quantize_enabled_op_types.empty()) {
+  supported_op_types = quantize_enabled_op_types;
+}
+auto *op = pattern -> NewNode(op_repr()) -> assert_is_ops(supported_op_types);
+return op;
```
Comment (arogowie-intel) [inline paddle/fluid/framework/ir/graph_pattern_detector.cc]: You have a compile error here. There is duplication of 2255 line. And broken formatting.

**Review by wozna — APPROVED** [review]
LGTM

File: paddle/fluid/framework/ir/graph_pattern_detector.cc (hunk 1 above)
Comment (lidanqing-vv) [reply paddle/fluid/framework/ir/graph_pattern_detector.cc]: Hi, Thank you. I haved talke with Joanna, and removed those changes that she has add in her merged PR. Now I m just waiting for CIs to pass. Locally UT passed

Comment (lidanqing-vv) [reply paddle/fluid/framework/ir/graph_pattern_detector.cc]: @arogowie-intel  Hi it passed CIs, please review again. Thanks.

**Comment by lidanqing-vv** [comment]
@jczaja Hi, Please review this PR.

**Review by jczaja — APPROVED** [review]
LGTM

**Pull request merged**